price-feeder/oracle/client/client.go

```diff
@@ -0,0 +1,274 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+	tmjsonclient "github.com/tendermint/tendermint/rpc/jsonrpc/client"
+	umeeapp "github.com/umee-network/umee/app"
+	umeeparams "github.com/umee-network/umee/app/params"
+)
+
+type (
+	CosmosChain struct {
```

Comment:
- I would combine `CosmosChain` and `OracleClient` into a single structure.
- I would remove `Ctx` and `Factory` from the structure as that's a bit of a code-smell
  - `Context` and `TxFactory` should probably be generated on the fly via a method on `CosmosChain`. E.g. `CosmosChain#CreateContext` and `CosmosChain#CreateTxFactory`